PatternPal/PatternPal.Extension/Commands/NavigateCommand.cs

```diff
@@ -37,7 +48,142 @@ public class NavigateCommand< T > : ICommand
+                            (IVsUIShell)Package.GetGlobalService(typeof(IVsUIShell))))
+            {
+                string filePath = string.Empty;
+                // If there are no projects in the solution create one.
+                if (dte.Solution.Projects.Count == 0)
+                {
+                    string projectName = "NewProject";
+                    string projectPath = System.IO.Path.Combine(dte.Solution.FullName, projectName);
+                    Project csTemplateProject =
+                        dte.Solution.AddFromTemplate(
+                            "ConsoleApplication",
+                            projectPath,
+                            projectName,
+                            false);
+
+                    string fileName = "NewFile.cs";
+                    string fileContent = "";
+                    filePath =
+                        System.IO.Path.Combine(
+                            System.IO.Path.GetDirectoryName(csTemplateProject.FullName),
+                            fileName);
+                    System.IO.File.WriteAllText(filePath, fileContent);
+                    csTemplateProject.ProjectItems.AddFromFile(filePath);
+                }
+                else
+                {
+                    Project project = dte.Solution.Projects.Item(1);
+                    string fileName = "NewFile.cs";
+                    string fileContent = "";
+                    filePath =
+                        System.IO.Path.Combine(
+                            System.IO.Path.GetDirectoryName(project.FullName),
+                            fileName);
+                    System.IO.File.WriteAllText(filePath, fileContent);
+                    project.ProjectItems.AddFromFile(filePath);
+                }
+                dte.ItemOperations.OpenFile(filePath);
+                stepByStepModel.FilePath = filePath;
+                return stepByStepModel;
+            }
+            else
+            {
+                // Call what you would do otherwise when there is no solution like in
+                StepByStepInstructionsViewModel newViewModel = 
+                    CreateNewWorkFile(stepByStepModel);
+                return newViewModel;
+            }
+        }
+
+        /// <summary>
+        /// Prompts the user to provide a filepath and creates a new .cs file and opens it.
+        /// </summary>
+        /// <param name="stepByStepModel"></param>
+        /// <returns></returns>
+        public StepByStepInstructionsViewModel CreateNewWorkFile(
+            StepByStepInstructionsViewModel stepByStepModel)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            // Obtain the currently running visual studio instance.
+            DTE dte = (DTE)Package.GetGlobalService(typeof(SDTE));
+
+            // Save location for project provided by user
+            string folderPath = string.Empty;
+            using (FolderBrowserDialog fdb = new FolderBrowserDialog())
```

Comment:
Why does the user not get this option when choosing to add the file to the solution? It would be better if that behavior is the same.

Reply:
The CreateNewWorkFile is now called everytime one wants to create a new work file.